ext/pdo_pgsql/pdo_pgsql.c

```diff
@@ -53,6 +57,93 @@ zend_module_entry pdo_pgsql_module_entry = {
 ZEND_GET_MODULE(pdo_pgsql)
 #endif
 
+/* Escape an identifier for insertion into a text field	*/
+PHP_METHOD(PdoPgsql, escapeIdentifier)
+{
+	zend_string *from = NULL;
+	char *tmp;
+	pdo_dbh_t *dbh;
+	pdo_pgsql_db_handle *H;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "S", &from) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	dbh = Z_PDO_DBH_P(ZEND_THIS);
+	PDO_CONSTRUCT_CHECK;
+	PDO_DBH_CLEAR_ERR();
+
+	/* Obtain db Handle */
+	H = (pdo_pgsql_db_handle *)dbh->driver_data;
+	if (H->server == NULL) {
+		zend_throw_error(NULL, "PostgreSQL connection has already been closed");
+		RETURN_THROWS();
+	}
+
+	tmp = PQescapeIdentifier(H->server, ZSTR_VAL(from), ZSTR_LEN(from));
```

Comment:
At last, I found https://doxygen.postgresql.org/fe-exec_8c_source.html#l04123 based on which it looks like we can throw an exception (thus, I removed Dan's TODO comment)

Reply:
Shouldn't we retrieve the PGSQL error message when we throw the error?

Reply:
Besides the OOM error, I've realized there is a `incomplete multibyte character` error case as well. It is not a programmatic error, is it? 🤔 But yes, it sounds like a good ide to extend our error message with the one provided by PgSQL

Reply:
I think that it can be considered a programming error to not check that you are sending valid data to the DB?

Reply:
Hmm, I'm not 100% sold, but let's go with the exception for now. At least we don't have the false return type! yay!